### DISCUSSION TITLE
Skip or prevent API fetching on build

### DISCUSSION THREAD

**Discussion by tilvan** [vercel/next.js#28362] [Help]
I use `getInitialProps` in `_app.ts`. Where I fetch API for initial data.
And my question is how I can skip API fetching on build.
I want to fetch API only on http request.

```
import React from 'react';
import App, { AppContext, AppInitialProps, AppProps } from 'next/app';

import { Provider } from 'react-redux';
import store from 'stores/store';
import { AppPropsProvider } from 'helpers/pagePropsContex';
import { getMenuServerItems } from 'services/MainMenuService';

import '../styles/global.scss';

const NextApp = (props: AppProps<AppInitialProps>) => {
  const { Component, pageProps } = props;

  return (
    <Provider store={store}>
      <AppPropsProvider value={pageProps}>
        <Component {...pageProps} />
      </AppPropsProvider>
    </Provider>
  );
};

NextApp.getInitialProps = async (appContext: AppContext) => {
  const pageProps = await App.getInitialProps(appContext);
  const menuItemsResponse = await getMenuServerItems();
  const { menuItems = [] } = menuItemsResponse.data;

  const nextProps = {
    pageProps: {
      ...pageProps.pageProps,
      ...{
        menuItems,
      },
    },
  };

  return nextProps;
};

export default NextApp;
```

**Comment by oliverloops** [comment]
Hi @tilvan, since Next.js 9.3 release is recommended to use more data fetching granular alternatives  like `getServerSideProps` and `getStaticProps` instead of `getInitialProps`. When `getInitialProps` is used in your custom  `_app.ts` this tends to run on the initial load, you could use `getServerSideProps` instead in order to fetch your API on server side and avoid to run always on build time.

You can find the whole reference here: https://nextjs.org/docs/api-reference/data-fetching/getInitialProps

 I'll hope this works. ✌️

**Reply by tilvan** [reply]
Hi @oliverloops!
Thx fo response. Of course I already read a docs.
I used `getInitialProps` because there I fetch data for all of pages(in `pages` folder) once.
In other case, if I will use `getServerSideProps` I must copy request snippet to each other `/pages/*` files.
It might be 1, 10 or 50, 100 pages in project.
